core-bundle/src/Menu/FrontendMenuBuilder.php

```diff
@@ -263,35 +300,27 @@ private function generateUri(PageModel $pageModel, ItemInterface $menuItem): ?st
         }
     }
 
-    private function populateMenuItem(ItemInterface $item, Request $request, PageModel $page, ?string $href, bool $hasSubmenu, array $options = []): void
+    private function populateMenuItem(ItemInterface $item, PageModel $page, array $options = []): void
     {
-        /** @var PageModel|int|null $currentPage */
-        $currentPage = $request->attributes->get('pageModel');
-
-        // Support ESI requests
-        if (null !== $currentPage && !$currentPage instanceof PageModel) {
-            /** @var PageModel|null $currentPage */
-            $currentPage = $this->pageModelAdapter->findByPk($currentPage);
-        }
+        $request = $this->requestStack->getCurrentRequest();
+        $activePage = $this->getActivePage();
 
         $extra = $page->row();
-        $isTrail = $currentPage && \in_array($page->id, $currentPage->trail, false);
-
-        $item->setUri($href);
+        $isTrail = $activePage && \in_array($page->id, $activePage->trail, false);
 
         // Use the path without query string to check for active pages (see #480)
         $path = ltrim($request->getPathInfo(), '/');
 
-        $isActive = $currentPage
-            && $href === $path
+        $isActive = $activePage
```

Comment:
I don't like the renaming from $currentPage to $activePage because we use the term "active" for the page that is marked _active_ in the menu. I used $currentPage because the page model comes from the current request, but $requestPage would be also fine.

Reply:
Yes, I originally renamed it for some other reason - but in the current implementation it would not be necessary. However, I still think `$activePage` is more accurate, since it denotes the active page - where as the "current" page is imho the page that is currently processed when building the menu.

Reply:
Nevertheless, I have renamed it back to `$currentPage` :)